Lib/dis.py

```diff
@@ -376,8 +376,9 @@ def _disassemble(self, lineno_width=3, mark_as_current=False, offset_width=4):
         fields = []
         # Column: Source code line number
         if lineno_width:
-            if self.starts_line is not None:
-                lineno_fmt = "%%%dd" % lineno_width
+            if self.starts_line is not False:
```

Comment:
Rather than overload `starts_line` in this way, maybe split into two attributes, `starts_line` which is just a boolean and `line_number` which is the actual line number.
The logic is then clearer:
```Py
if self.starts_line:
    lineno_fmt = "%%%dd" if self.line_number is not None else "%%%ds"
    lineno_fmt = lineno_fmt % lineno_width
    fields.append(lineno_fmt % self.line_number)
else:
    fields.append(' ' * lineno_width)
```
```